sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java

```diff
@@ -517,19 +525,23 @@
      * @throws HttpResponseException If {@code key} is an empty string.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<ConfigurationSetting>> deleteSettingWithResponse(ConfigurationSetting setting) {
-        return withContext(context -> deleteSetting(setting, context));
+    public Mono<Response<ConfigurationSetting>> deleteSettingWithResponse(String key, String label, String etag) {
+        return withContext(context -> deleteSetting(key, label, etag, context));
     }
 
-    Mono<Response<ConfigurationSetting>> deleteSetting(ConfigurationSetting setting, Context context) {
+    Mono<Response<ConfigurationSetting>> deleteSetting(String key, String label, String etag, Context context) {
         // Validate that setting and key is not null. The key is used in the service URL so it cannot be null.
-        validateSetting(setting);
+        if (key == null) {
```

Comment:
Based on the spec, this should be Objects.requireNotNull and throw NullPointerException.
https://azure.github.io/azure-sdk/java_implementation.html